enhancements/kube-apiserver/stability-point-to-point-network-check.md

```diff
@@ -0,0 +1,447 @@
+
+A point to point connection is between a source pod and a target endpoint.
+
+Component operators will create a `PodNetworkConnectivityCheck` resource per point-to-point connection. If an operator 
+is not managing the source pod directly, but via a workload controller (such as via a deployment or daemonset), the 
+operator will observe the workload resources to determine the list of pods and create a `PodNetworkConnectivityCheck` 
+resource for each target endpoint per pod. Operators that directly manage a source pod (such as operators that manage 
+static-pods) will create `PodNetworkConnectivityCheck` resources for each target endpoint when creating the source pod.
+
+### 2. Detailed status, probably some kind of latency information.
+
+The `PodNetworkConnectivityCheck` will maintain a log of the actions performed during a check, recording if the actions
+were successful, a reason for success or failure, latency of performing the action and some human readable message.
+
+**LogEntry:**
+```go
+// LogEntry records events
+type LogEntry struct {
+	// Start time of check action.
+	Start metav1.Time `json:"time,omitempty"`
+	// Status indicates if the action performed by the check was successful or not.
+	Success bool `json:"status"`
+	// Reason for status in a machine readable format.
+	Reason LogEntryReason `json:"reason"`
+	// Message explaining status in a human readable format.
+	Message string `json:"message"`
+	// Latency records how long the action mentioned in the entry took.
+	Latency time.Duration `json:"latency"`
+}
+``` 
+
+The start and end time of detected outages will be logged separately from the individual check actions.
+
+### 3. It should be possible to clean up garbage after a couple hours of inactivity.
+
+A controller to prune `PodNetworkConnectivityCheck` resources, will be provided for inclusion into source pod operators.
+
+A `PodNetworkConnectivityCheck` resource would be pruned after 48 hours of inactivity.
+
+Tools updating `PodNetworkConnectivityCheck` log entries should simultaneously delete the oldest log entry to ensure
+there are never more that 20 entries.
+
+### 4. Don't delete aggressively, on upgrades, we have different endpoints
+
+Source pods managed by workload controllers will have new `PodNetworkConnectivityCheck` resources to go along with their
+new names. The old `PodNetworkConnectivityCheck` resource will eventually go stale and be cleaned up by the provided
+pruning controller. 
+
+Directly managed pods whose names do not change (such as static pods) will share `PodNetworkConnectivityCheck` resources
+across revisions and upgrades.
+
+### 5. Create a binary that can take multiple destinations paired with instances to write to.
+
+### 6. Use that binary in containers included in each kas and oas (or other interesting) pod.
+
+The binary will:
+
+1. List all `PodNetworkConnectivityCheck` resource in the same namespace, where `spec.sourcePodName` matches the pod the
```

Comment:
You are going to be accessing etcd over the host network, so this name is confusing.  It would probably be more appropriate to scope this more the core control plane - I don't see how this would be appropriate for general use since it's not tied to pod lifecycle or monitoring.

Reply:
> You are going to be accessing etcd over the host network, so this name is confusing. It would probably be more appropriate to scope this more the core control plane - I don't see how this would be appropriate for general use since it's not tied to pod lifecycle or monitoring.

we need a unique identifier for a particular instance of a particular control plane binary.  Right now, the simplest one we have a pod name.  It's unique, they all have it, we can discover them all, and its injectable into the container cmd.